### ISSUE TITLE
Serving model in 8bit mode fails

### ISSUE THREAD

**Issue by minosvasilias** [eth-sri/lmql#94]
Attempting to serve a HF model in 8bit mode as introduced here (https://github.com/eth-sri/lmql/pull/46) fails for me with the following error:

```
lmql serve-model HuggingFaceH4/starchat-beta --cuda --dtype 8bit

[Loading HuggingFaceH4/starchat-beta with AutoModelForCausalLM.from_pretrained(HuggingFaceH4/starchat-beta, 'torch_dtype': '8bit', 'load_in_8bit': True, 'device_map': 'auto')]
======== Running on http://localhost:8080 ========
(Press CTRL+C to quit)
Exception in thread Thread-1 (worker):
Traceback (most recent call last):
  File "/home/username/.pyenv/versions/3.11.4/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/home/username/.pyenv/versions/3.11.4/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/home/username/.pyenv/versions/3.11.4/lib/python3.11/site-packages/lmql/models/lmtp/lmtp_inference_server.py", line 257, in worker
    model = AutoModelForCausalLM.from_pretrained(self.model_identifier, **self.model_args)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/username/.pyenv/versions/3.11.4/lib/python3.11/site-packages/transformers/models/auto/auto_factory.py", line 484, in from_pretrained
    return model_class.from_pretrained(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/username/.pyenv/versions/3.11.4/lib/python3.11/site-packages/transformers/modeling_utils.py", line 2633, in from_pretrained
    raise ValueError(
ValueError: `torch_dtype` can be either `torch.dtype` or `"auto"`, but received 8bit
```

See [transformers/modeling_utils.py](https://github.com/huggingface/transformers/blob/d7389cd20168052e5fc7abe0cf31cd1eb960fbc9/src/transformers/modeling_utils.py#L2633).

**Comment by lbeurerkellner** [comment]
Do you have `bitsandbytes` installed?

**Comment by minosvasilias** [comment]
Hey @lbeurerkellner , apologies for the delayed reply! Yes, `bitsandbytes` is installed.

Also, to confirm, this is happening on both Python version `3.10` (as indicated in the Readme) as well as `3.11`, and i have gotten the same error with several different models i have tried.

**Comment by minosvasilias** [comment]
I opened a PR with a proposed fix here: #97

**Comment by lbeurerkellner** [comment]
Thanks, I just merged the PR.
